src/main/java/net/pms/network/RequestV2.java

```diff
@@ -671,79 +671,43 @@ public ChannelFuture answer(
 			}
 		} else if (method.equals("POST") && (argument.contains("MS_MediaReceiverRegistrar_control") || argument.contains("mrr/control"))) {
 			output.headers().set(HttpHeaders.Names.CONTENT_TYPE, "text/xml; charset=\"utf-8\"");
-			response.append(HTTPXMLHelper.XML_HEADER);
-			response.append(CRLF);
-			response.append(HTTPXMLHelper.SOAP_ENCODING_HEADER);
-			response.append(CRLF);
 
+			StringBuilder payload = new StringBuilder();
 			if (soapaction != null && soapaction.contains("IsAuthorized")) {
-				response.append(HTTPXMLHelper.XBOX_360_2);
-				response.append(CRLF);
+				payload.append(HTTPXMLHelper.XBOX_360_2);
+				payload.append(CRLF);
 			} else if (soapaction != null && soapaction.contains("IsValidated")) {
-				response.append(HTTPXMLHelper.XBOX_360_1);
-				response.append(CRLF);
+				payload.append(HTTPXMLHelper.XBOX_360_1);
+				payload.append(CRLF);
 			}
 
-			response.append(HTTPXMLHelper.SOAP_ENCODING_FOOTER);
-			response.append(CRLF);
+			response.append(createResponse(payload.toString()));
 		} else if (method.equals("POST") && argument.endsWith("upnp/control/connection_manager")) {
 			output.headers().set(HttpHeaders.Names.CONTENT_TYPE, "text/xml; charset=\"utf-8\"");
 
 			if (soapaction != null && soapaction.contains("ConnectionManager:1#GetProtocolInfo")) {
-				response.append(HTTPXMLHelper.XML_HEADER);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.SOAP_ENCODING_HEADER);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.PROTOCOLINFO_RESPONSE);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.SOAP_ENCODING_FOOTER);
-				response.append(CRLF);
+				response.append(createResponse(HTTPXMLHelper.PROTOCOLINFO_RESPONSE));
 			}
 		} else if (method.equals("POST") && argument.endsWith("upnp/control/content_directory")) {
 			output.headers().set(HttpHeaders.Names.CONTENT_TYPE, "text/xml; charset=\"utf-8\"");
 
 			if (soapaction != null && soapaction.contains("ContentDirectory:1#GetSystemUpdateID")) {
-				response.append(HTTPXMLHelper.XML_HEADER);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.SOAP_ENCODING_HEADER);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.GETSYSTEMUPDATEID_HEADER);
-				response.append(CRLF);
-				response.append("<Id>").append(DLNAResource.getSystemUpdateId()).append("</Id>");
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.GETSYSTEMUPDATEID_FOOTER);
-				response.append(CRLF);
-				response.append(HTTPXMLHelper.SOAP_ENCODING_FOOTER);
-				response.append(CRLF);
+				StringBuilder payload = new StringBuilder();
+				payload.append(HTTPXMLHelper.GETSYSTEMUPDATEID_HEADER);
+				payload.append(CRLF);
```

Comment:
@drakulis It could be more readable to chain the `CRLF`s too since they take up a lot of space, e.g. `payload.append(HTTPXMLHelper.GETSYSTEMUPDATEID_HEADER).append(CRLF);` what do you think?
This branch is already an improvement anyway

Reply:
Good idea. Done.

If you ask me I would completely remove those `CRLF`. Sure, they may provide more readable string during debug but overall they blur the code. When I finish refactoring this class I will try to implement request / response objects as a Java object to get rid of playing around with concatenation of the strings. I will use JAXB to serialize / deserialize payloads like I already done in SamsungBookmark.

@SubJunk If you are fine with current changes please merge this PR. I will take care of next step from the list then.

Reply:
There may be some requirement in DLNA with line-breaks? I would have to look it up. Either way it would be good to keep this a like-for-like refactoring, and make a separate issue for removing line-breaks afterwards, if we are focusing on keeping the PRs/branches separated for fast review. I'm happy to merge this if my last comment can be addressed (the double CRLF)

Reply:
DLAN protocol is based on SOAP. So payload is a regular XML. Line breaks and other white characters have no impact on it. If you take a look on payloads submitted from the devices to UMS you will notice that in mosts cases payload is a one-line string.

Reply:
@SubJunk It's like @drakulis says, the CRLF's have no consequence and have either been added for human readability/debugging or lack of knowledge.

Reply:
Cool it's good to be sure of that :)